src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1553,6 +1641,23 @@ public void ReorderFiles(ImmutableArray<string> filePaths)
 
                 ClearAndZeroCapacity(_documentsAddedInBatch);
 
+                // Document renaming...
+                foreach (var (documentId, oldPath, newPath) in _documentsRenamedInbatch)
+                {
+                    if (documentFileNamesAdded.Contains(oldPath))
```

Comment:
How would that happen? It looked like you already handled that in the earlier case?